core/src/main/scala/pureconfig/error/ConfigReaderFailure.scala

```diff
@@ -5,14 +5,152 @@ package pureconfig.error
 
 import com.typesafe.config.ConfigValue
 
-sealed abstract class ConfigReaderFailure
-final case object CannotConvertNull extends ConfigReaderFailure
-final case class CannotConvert(value: String, toTyp: String, because: String) extends ConfigReaderFailure
-final case class CollidingKeys(key: String, existingValue: String) extends ConfigReaderFailure
-final case class KeyNotFound(key: String) extends ConfigReaderFailure
-final case class UnknownKey(key: String) extends ConfigReaderFailure
-final case class WrongType(foundTyp: String, expectedTyp: String) extends ConfigReaderFailure
-final case class WrongTypeForKey(typ: String, expectedTyp: String, key: String) extends ConfigReaderFailure
-final case class ThrowableFailure(throwable: Throwable) extends ConfigReaderFailure
-final case class EmptyStringFound(typ: String) extends ConfigReaderFailure
-final case class NoValidCoproductChoiceFound(value: ConfigValue) extends ConfigReaderFailure
\ No newline at end of file
+/**
+ * The physical location of a ConfigValue, represented by a file name and a line
+ * number
+ *
+ * @param filename the complete pathname to the file that contains a given
```

Comment:
If the config file is inside of a JAR does this become some useful value like:
`path/to/the.jar#resource/path.conf`?